pkg/providers/imagefamily/nodeimage.go

```diff
@@ -0,0 +1,84 @@
+/*
+Portions Copyright (c) Microsoft Corporation.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package imagefamily
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/karpenter-provider-azure/pkg/apis/v1alpha2"
+	"github.com/Azure/karpenter-provider-azure/pkg/operator/options"
+	"sigs.k8s.io/karpenter/pkg/scheduling"
+)
+
+type NodeImage struct {
+	ID           string
+	Requirements scheduling.Requirements
+}
+
+type NodeImages []NodeImage
+
+type NodeImageProvider interface {
+	List(ctx context.Context, nodeClass *v1alpha2.AKSNodeClass) (NodeImages, error)
+}
+
+func (p *Provider) List(ctx context.Context, nodeClass *v1alpha2.AKSNodeClass) (NodeImages, error) {
+	supportedImages := getSupportedImages(nodeClass.Spec.ImageFamily)
+	if options.FromContext(ctx).UseSIG {
+		nodeImages := NodeImages{}
+		retrievedLatestImages, err := p.NodeImageVersions.List(ctx, p.location, p.subscription)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, nextSupportedImage := range supportedImages {
```

Comment:
minor, style: 
```suggestion
		for _, supportedImage := range supportedImages {
```

Calling it `nextSupportedImage` is sorta weird?
Applies below too.

Reply:
Changed in both locations